File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0
    y_max, x_max = 0, 0
    for i in range(len(sizes)):
        if sizes[i][0] < sizes[i][1]:
            sizes[i][0], sizes[i][1] = sizes[i][1], sizes[i][0]
        y_max = max(y_max, sizes[i][0])
        x_max = max(x_max, sizes[i][1])
    answer = x_max * y_max
    return answer

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.ArrayList;
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        ArrayList<Integer> w = new ArrayList<>();
        ArrayList<Integer> h = new ArrayList<>();
        for (int i = 0; i < sizes.length; i++) {
            if (sizes[i][0] < sizes[i][1]) {
                w.add(sizes[i][1]);
                h.add(sizes[i][0]);
            } else {
                w.add(sizes[i][0]);
                h.add(sizes[i][1]);
            }
        }
        w.sort(Integer::compareTo);
        h.sort(Integer::compareTo);
        answer = w.get(w.toArray().length-1) * h.get(h.toArray().length-1);
        return answer;
    }
}